lib/k8s/conn/auth/exec_worker.ex

```diff
@@ -0,0 +1,215 @@
+defmodule K8s.Conn.Auth.ExecWorker do
+  @moduledoc false
+  use GenServer
+
+  alias K8s.Conn.Error
+
+  defmodule State do
+    @moduledoc """
+    The state of the exec worker
+    """
+
+    defstruct [
+      :command,
+      :env,
+      args: [],
+      token: nil,
+      expiration_timestamp: nil,
+      refresh_interval: 60_000,
+      timer: nil
+    ]
+
+    @type t :: %__MODULE__{
```

Comment:
For further development it might help to have a hint to where this structure is coming from, wdyt?

```suggestion
    @typedoc """ 
    Simplified version of [ExecConfig](https://kubernetes.io/docs/reference/config-api/kubeconfig.v1/#ExecConfig)
    """
    @type t :: %__MODULE__{
```

Reply:
Perfect, thank you. I'll add some more typedocs for state